File: Ems Backend/src/main/java/com/cms/employeemanagement/service/LeaveService.java
package com.cms.employeemanagement.service;

import java.util.List;
import com.cms.employeemanagement.dto.LeaveBalanceDto;

import com.cms.employeemanagement.dto.LeaveRequestDto;
import com.cms.employeemanagement.entity.LeaveRequest;

public interface LeaveService {

    LeaveRequest applyLeave(LeaveRequestDto request);

    List<LeaveRequest> getMyLeaves(String email);

    List<LeaveRequest> getAllLeaves();

    LeaveRequest approveLeave(Long id);

    LeaveRequest rejectLeave(Long id);

    LeaveRequest cancelLeave(Long id);
    LeaveBalanceDto getLeaveBalance(String email);
}

File: Frontend - Ems/src/services/leaveService.js
import api from "../api/axios";

// Employee
export const applyLeave = (leaveData) =>
    api.post("/leaves/apply", leaveData);

export const getMyLeaves = (email) =>
    api.get(`/leaves/my/${email}`);

export const cancelLeave = (id) =>
    api.put(`/leaves/${id}/cancel`);

// HR/Admin
export const getAllLeaves = () =>
    api.get("/leaves");

export const approveLeave = (id) =>
    api.put(`/leaves/${id}/approve`);

export const rejectLeave = (id) =>
    api.put(`/leaves/${id}/reject`);

// Leave Balance
export const getLeaveBalance = (email) =>
    api.get(`/leaves/balance/${email}`);